### PR TITLE
A variety of doc updates

### PR DESCRIPTION
**Pull request by bangerth**
The primary motivation was to make sure that every `@ingroup` has a corresponding `@defgroup` statement.

Also a variety of other issues that came up in the context of looking through our documentation pages.

### CODE REVIEW

**Comment by bangerth** [comment]
Fixes #410

**Comment by bangerth** [comment]
The one failing test doesn't have anything to do with this PR -- it fails because of a bad prior merge.

File: include/aspect/material_model/diffusion_dislocation.h [hunk 1]
```diff
@@ -39,8 +39,8 @@ namespace aspect
      * effective viscosity functions describing diffusion and dislocation creep:
      * \\[v_{eff} = \\left( \\frac{1}{v_{eff}^{diff}} + \\frac{1}{v_{eff}^{dis} \\right)^{-1}\\]
      * where
-     * \\[v_{eff}^{diff} = A_{diff}^-1 \\exp\\left(\frac{E_{diff} + PV_{diff}}{RT}\\right)\\]
-     * \\[v_{eff}^{dis} =  A_{dis}^{\\frac{-1}{n_dis}} \\dot{\\varepsilon}^{\frac{1-n}{n}} \\exp\\left(\frac{E_{diff} + PV_{diff}}{n_{dis}RT}\\right)\\]
+     * \\[v_{eff}^{diff} = A_{diff}^-1 \\exp\\left(\\frac{E_{diff} + PV_{diff}}{RT}\\right)\\]
```
Comment (tjhei) [inline include/aspect/material_model/diffusion_dislocation.h]: All these formulas are incorrect markup for doxygen. Multline formulas begin with \f[, all "\\" should be single "\", there is a missing closing "}" in line 40, etc..
Please check by running doxygen and looking at the output.

Comment (bangerth) [inline include/aspect/material_model/diffusion_dislocation.h]: Good spot. Updated now -- even looks good in the output.

**Pull request merged**